internal/internal_workflow_client.go

```diff
@@ -1036,6 +1036,81 @@ func (wc *WorkflowClient) GetWorkerTaskReachability(ctx context.Context, options
 	return converted, nil
 }
 
+// DescribeTaskQueueEnhanced returns information about the target task queue, broken down by Build Id:
+//   - List of pollers
+//   - Workflow Reachability status
+//   - Backlog info for Workflow and/or Activity tasks
+//
+// WARNING: Worker versioning is currently experimental, and requires server 1.24+
+func (wc *WorkflowClient) DescribeTaskQueueEnhanced(ctx context.Context, options *DescribeTaskQueueEnhancedOptions) (TaskQueueDescription, error) {
```

Comment:
It is a bit confusing to have half of versioning client code in this file and half in another, but I understand why. There is an argument to be made for a `VersioningClient()` similar to our `ScheduleClient()`, but I'm ok w/ this since it's only 3 methods. No need to change anything, just thinking out loud.